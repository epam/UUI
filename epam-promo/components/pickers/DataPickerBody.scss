@use  '../../assets/styles/index' as *;

.body {
    flex: 1 1 auto;
}

.modal {
    border: 0;
}

// TBD: replace with search prop when it will be done
.search {
    &:global(.uui-input-box) {
        border: 0;
        box-shadow: none;

        &:global(.uui-focus) {
            border-color: transparent;
            box-shadow: none;
        }
    }
}

.checkbox {
    padding: 6px 0;
}

.search-wrapper {
    border-bottom: 1px solid $gray40;
    background: white;
    padding: 0 4px;
}

.search-size-24, .search-size-30, .search-size-36, .search-size-42 {
    padding: 0 5px;
}

.no-found-size-24 {
    padding-top: 20px;
    padding-bottom: 20px;
}

.no-found-size-30, .no-found-size-36, .no-found-size-42 {
    padding-top: 23px;
    padding-bottom: 23px;
}