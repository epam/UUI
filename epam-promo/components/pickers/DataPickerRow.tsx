import * as React from 'react';
import * as ReactDOM from 'react-dom';
import isEqual from 'lodash.isequal';
import { DataRowProps, DataColumnProps, FlexRowProps, uuiMod, uuiMarkers } from '@epam/uui';
import { FlexRow } from '@epam/uui-components';
import { FlexSpacer } from '../layout';
import { DataTableCell } from '../tables';
import { IconButton } from '../buttons';
import * as css from './DataPickerRow.scss';
import * as tickIcon_24 from '@epam/assets/icons/common/notification-done-24.svg';
import * as tickIcon_18 from '@epam/assets/icons/common/notification-done-18.svg';
import * as tickIcon_12 from '@epam/assets/icons/common/notification-done-12.svg';

export interface DataPickerRowProps<TItem, TId> extends DataRowProps<TItem, TId> {
    renderItem(item: TItem, rowProps: DataRowProps<TItem, TId>): React.ReactNode;
    padding?: '12' | '24';
    size?: 'none' | '24' | '30' | '36' | '42' | '48' | '60';
    borderBottom?: 'none' | 'gray20';
    alignActions?: 'top' | 'center';
}

export class DataPickerRow<TItem, TId> extends React.Component<DataPickerRowProps<TItem, TId>> {

    private getIcon = (size: string) => {
        switch (size) {
            case '24': return tickIcon_12;
            case '30': return tickIcon_18;
            case '36': return tickIcon_18;
            case '42': return tickIcon_24;
            default: return tickIcon_18;
        }
    }

    rowNode: React.RefObject<any> = React.createRef();
    rowDOMNode: Element | Text = null;

    column: DataColumnProps<TItem> =
        {
            key: 'name',
            grow: 1,
            render: (item, rowProps) => <div key={ rowProps.id }  className={ css.renderItem }>
                { this.props.renderItem(item, rowProps) }
                <FlexSpacer />
                { (rowProps.isChildrenSelected || rowProps.isSelected) && <div className={ css.iconWrapper }>
                    <IconButton icon={ this.getIcon(this.props.size) } color={ rowProps.isChildrenSelected ? 'gray60' : 'blue' } />
                </div> }
            </div>,
        };

    componentDidMount() {
        this.rowDOMNode = ReactDOM.findDOMNode(this.rowNode.current);
        if (this.props.onFocus) {
            this.rowDOMNode?.addEventListener('mouseenter', this.handleMouseEnter);
        }
    }

    componentWillUnmount() {
        this.rowDOMNode?.removeEventListener('mouseenter', this.handleMouseEnter);
    }

    handleMouseEnter = (e: any) => {
        this.props.onFocus(this.props.index);
    }

    shouldComponentUpdate(nextProps: DataRowProps<TItem, TId> & FlexRowProps) {
        const eq = isEqual(this.props, nextProps);
        return !eq;
    }

    render() {
        const clickHandler = this.props.onSelect || this.props.onFold || this.props.onCheck;
        return <FlexRow
            onClick={ () => clickHandler && clickHandler(this.props)}
            rawProps={ {
                role: 'option',
                'aria-setsize': this.props.columns?.length == undefined ? undefined : this.props.columns.length + 1,
                'aria-posinset': this.props.index,
                'aria-checked': this.props.isChecked,
                ...(this.props.isSelectable && { 'aria-selected': this.props.isSelected }),
                ...this.props.rawProps,
            } }
            ref={ this.rowNode }
            cx={ [
                css.pickerRow,
                clickHandler && uuiMarkers.clickable,
                clickHandler && this.props.isFocused && uuiMod.focus,
                this.props.cx,
            ] }
        >
            <DataTableCell
                key='name'
                size={ this.props.size || '36' }
                padding={ this.props.padding || '24' }
                isFirstColumn={ true }
                isLastColumn={ false }
                column={ this.column }
                rowProps={ this.props }
                alignActions={ this.props.alignActions || 'top' }
            />
        </FlexRow>;
    }
}
