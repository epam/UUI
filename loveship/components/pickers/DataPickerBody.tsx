import React from 'react';
import ReactDOM from 'react-dom';
import isEqual from 'lodash.isequal';
import css from './DataPickerBody.scss';
import cx from 'classnames';
import { Lens, DataSourceState, IEditable, DataSourceListProps, isMobile } from '@epam/uui';
import { FlexCell } from '@epam/uui-components';
import { i18n } from "../../i18n";
import * as types from '../types';
import { SearchInput } from '../inputs';
import { FlexRow, VirtualList } from '../layout';
import { Text } from '../typography';

export type DataPickerBodyProps<TItem, TId> = DataSourceListProps & IEditable<DataSourceState> & {
    showSearch?: boolean | 'auto'
    showSelectedRows?: boolean;
    maxHeight?: number;
    rows: React.ReactNode[];
    renderNotFound?: () => React.ReactNode;
    editMode?: 'dropdown' | 'modal';
    onKeyDown?(e: React.KeyboardEvent<HTMLElement>): void;
    scheduleUpdate?: () => void;
    searchSize?: types.ControlSize;
    search: IEditable<string>;
};

export class DataPickerBody<TItem, TId> extends React.Component<DataPickerBodyProps<TItem, TId>, any> {
    lens = Lens.onEditableComponent<DataSourceState>(this);
    searchLens = this.lens.prop('search');
    needFocusSearch: boolean = this.showSearch();

    componentDidUpdate(prevProps: DataPickerBodyProps<TItem, TId>) {
        if (this.needFocusSearch) {
            let body = ReactDOM.findDOMNode(this) as HTMLElement;
            body && body.getElementsByTagName('input')[0].focus({ preventScroll: true });
            this.needFocusSearch = false;
        }
        if (prevProps.rows.length !== this.props.rows.length || !isEqual(prevProps.value.checked, this.props.value.checked)) {
            this.props.scheduleUpdate && this.props.scheduleUpdate();
        }
    }

    showSearch() {
        let showSearch = this.props.showSearch;
        if (showSearch === 'auto') {
            showSearch = this.props.totalCount > 10;
        }

        return showSearch;
    }

    renderNoFound() {
        if (this.props.renderNotFound) {
            return this.props.renderNotFound();
        }

        return <FlexCell grow={ 1 } textAlign="center">
            <Text>{ i18n.dataPickerBody.noResultsMessage }</Text>
        </FlexCell>;
    }

    render() {
        const value = this.props.value;
        const searchSize = isMobile() ? "48" : (this.props.searchSize || "36");
        const searchClass = cx(css.searchWrapper, css[`search-size-${ searchSize }`]);

        return <>
            { this.showSearch() && (
                <div key="search" className={ searchClass }>
                    <FlexCell grow={ 1 }>
                        <SearchInput
                            cx={ css.search }
                            placeholder={ i18n.dataPickerBody.searchPlaceholder }
                            { ...this.searchLens.toProps() }
                            onKeyDown={ this.props.onKeyDown }
                            size={ searchSize }
                        />
                    </FlexCell>
                </div>
            ) }
            <FlexRow
                key="body"
                cx={ css.body }
                rawProps={ { style: { maxHeight: this.props.maxHeight } } }
                background="white"
            >
                { this.props.rowsCount > 0
                    ? <VirtualList
                        { ...this.lens.toProps() }
                        shadow="white"
                        role="listbox"
                        rows={ this.props.rows }
                        rowsCount={ this.props.rowsCount }
                        focusedIndex={ value && value.focusedIndex || 0 }
                    />
                    : this.renderNoFound()
                }
            </FlexRow>
        </>;
    }
}
