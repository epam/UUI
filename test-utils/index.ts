export * from './helpers';
export * from './mocks';
