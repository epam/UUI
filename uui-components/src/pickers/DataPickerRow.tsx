import * as React from 'react';
import isEqual from 'lodash.isequal';
import { DataRowProps, FlexRowProps, uuiMod } from '@epam/uui-core';
import { FlexRow } from '../index';

interface DataPickerRowProps<TItem, TId> extends DataRowProps<TItem, TId> {
    renderContent: () => React.ReactNode;
}

export class DataPickerRow<TItem, TId> extends React.Component<DataPickerRowProps<TItem, TId>> {
    rowNode = React.createRef<HTMLDivElement>();

    componentDidMount() {
        if (this.props.onFocus) {
            this.rowNode.current?.addEventListener('mouseenter', this.handleMouseEnter);
        }
    }

    componentWillUnmount() {
        this.rowNode.current?.removeEventListener('mouseenter', this.handleMouseEnter);
    }

    handleMouseEnter = () => {
        this.props.onFocus && this.props.onFocus(this.props.index);
    };

    shouldComponentUpdate(nextProps: DataRowProps<TItem, TId> & FlexRowProps) {
        return !isEqual(this.props, nextProps);
    }

    render() {
        const clickHandler = this.props.onClick || this.props.onSelect || this.props.onFold || this.props.onCheck;
        return (
            <FlexRow
                onClick={clickHandler && (() => clickHandler(this.props))}
                rawProps={{
                    role: 'option',
                    'aria-posinset': this.props.index + 1,
                    ...(this.props.checkbox?.isVisible && { 'aria-checked': this.props.isChecked }),
                    ...(this.props.isSelectable && { 'aria-selected': this.props.isSelected }),
                    ...this.props.rawProps,
                }}
                ref={this.rowNode}
                cx={[clickHandler && this.props.isFocused && uuiMod.focus, this.props.cx]}
            >
                {this.props.renderContent()}
            </FlexRow>
        );
    }
}
