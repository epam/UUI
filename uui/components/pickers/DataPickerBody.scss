@use  '../../assets/styles/index' as *;

.body {
    display: flex;
    flex-direction: column;
    flex: 1 1 auto;
    align-self: stretch;
    width: 100%;
}

.modal {
    border: 0;
}

// TBD: replace with search prop when it will be done
.search {
    &:global(.uui-input-box) {
        border: 0;
        box-shadow: none;
        background-color: transparent;

        &:global(.uui-focus) {
            border-color: transparent;
            box-shadow: none;
        }
    }
}

.checkbox {
    padding: 6px 0;
}

.search-wrapper {
    border-bottom: 1px solid var(--uui-divider);
    padding: 0 4px;
}

.search-size-24, .search-size-30, .search-size-36, .search-size-42 {
    padding: 0 5px;
}

.no-found-size-24 {
    padding-top: 20px;
    padding-bottom: 20px;
}

.no-found-size-30, .no-found-size-36, .no-found-size-42 {
    padding-top: 23px;
    padding-bottom: 23px;
}