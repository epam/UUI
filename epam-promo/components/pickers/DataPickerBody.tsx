import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as css from './DataPickerBody.scss';
import * as types from '../types';
import cx from 'classnames';
import { SearchInput } from '../inputs';
import { FlexRow, VirtualList } from '../layout';
import { Lens, DataSourceState, IEditable, DataSourceListProps, isMobile } from '@epam/uui';
import { FlexCell } from '@epam/uui-components';
import { Text } from '../typography';
import isEqual from 'lodash.isequal';
import { i18n } from "../../i18n";

export type DataPickerBodyProps<TItem, TId> = DataSourceListProps & IEditable<DataSourceState> & {
    showSearch?: boolean | 'auto'
    showSelectedRows?: boolean;
    maxHeight?: number;
    rows: React.ReactNode[];
    renderNotFound?: () => React.ReactNode;
    editMode?: 'dropdown' | 'modal';
    onKeyDown?(e: React.KeyboardEvent<HTMLElement>): void;
    scheduleUpdate?: () => void;
    searchSize?: types.ControlSize;
    search: IEditable<string>;
};

export class DataPickerBody<TItem, TId> extends React.Component<DataPickerBodyProps<TItem, TId>, any> {
    lens = Lens.onEditableComponent<DataSourceState>(this);
    searchLens = this.lens.prop('search');
    needFocusSearch: boolean = this.showSearch();


    componentDidUpdate(prevProps: DataPickerBodyProps<TItem, TId>) {
        if (this.needFocusSearch) {
            let body = ReactDOM.findDOMNode(this) as HTMLElement;
            body && body.getElementsByTagName('input')[0]?.focus({ preventScroll: true });
            this.needFocusSearch = false;
        }
        if (prevProps.rows.length !== this.props.rows.length || !isEqual(prevProps.value.checked, this.props.value.checked)) {
            this.props.scheduleUpdate && this.props.scheduleUpdate();
        }
    }

    showSearch() {
        let showSearch = this.props.showSearch;
        if (showSearch === 'auto') {
            showSearch = this.props.totalCount > 10;
        }

        return showSearch;
    }

    renderNoFound() {
        if (this.props.renderNotFound) {
            return this.props.renderNotFound();
        }

        return <FlexCell cx={ css[`no-found-size-${ this.props.searchSize || 36 }`] } grow={ 1 } textAlign="center">
            <Text size={ this.props.searchSize || '36' }>{ i18n.dataPickerBody.noRecordsMessage }</Text>
        </FlexCell>;
    }

    render() {
        const value = this.props.value;
        const searchSize = isMobile() ? "48" : (this.props.searchSize || "36");
        const searchClass = cx(css.searchWrapper, css[`search-size-${ searchSize }`]);

        return <>
            { this.showSearch() && (
                <div key="search" className={ searchClass }>
                    <FlexCell grow={ 1 }>
                        <SearchInput
                            cx={ css.search }
                            placeholder={ i18n.dataPickerBody.searchPlaceholder }
                            { ...this.searchLens.toProps() }
                            onKeyDown={ this.props.onKeyDown }
                            size={ searchSize }
                        />
                    </FlexCell>
                </div>
            ) }
            <FlexRow
                key="body"
                cx={ cx(css.body, css[this.props.editMode]) }
                rawProps={ { style: { maxHeight: this.props.maxHeight } } }
                background="white"
            >
                { this.props.rowsCount > 0
                    ? <VirtualList
                        { ...this.lens.toProps() }
                        shadow={ false }
                        rows={ this.props.rows }
                        rowsCount={ this.props.rowsCount }
                        focusedIndex={ value && value.focusedIndex || 0 }
                    />
                    : this.renderNoFound()
                }
            </FlexRow>
        </>;
    }
}
