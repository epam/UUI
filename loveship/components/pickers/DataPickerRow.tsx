import { DataRowProps, DataColumnProps, FlexRowProps, uuiMod, uuiMarkers } from '@epam/uui';
import React from 'react';
import * as ReactDOM from 'react-dom';
import isEqual from 'lodash.isequal';
import * as css from './DataPickerRow.scss';
import { DataTableRowMods } from '../tables';
import { IconButton } from '../buttons';
import * as tickIcon from '../icons/notification-done-18.svg';
import * as smallSizeIcon from '../icons/notification-done-12.svg';
import { FlexSpacer } from '../layout';
import { FlexRow } from '@epam/uui-components';
import { DataTableCell } from '../tables';

export interface DataPickerRowProps<TItem, TId> extends DataRowProps<TItem, TId>, DataTableRowMods {
    renderItem(item: TItem, rowProps: DataRowProps<TItem, TId>): React.ReactNode;
    alignActions?: 'top' | 'center';
}

export class DataPickerRow<TItem, TId> extends React.Component<DataPickerRowProps<TItem, TId>> {

    rowNode: React.RefObject<any> = React.createRef();
    rowDOMNode: Element | Text = null;

    column: DataColumnProps<TItem> =
        {
            key: 'name',
            grow: 1,
            render: (item, rowProps) => <div key={ rowProps.id } className={ css.renderItem }>
                { this.props.renderItem(item, rowProps) }
                <FlexSpacer />
                { (rowProps.isChildrenSelected || rowProps.isSelected) && <div className={ css.iconWrapper }>
                    <IconButton icon={ this.props.size === '24' ? smallSizeIcon : tickIcon } color={ rowProps.isChildrenSelected ? 'night400' : 'sky' } />
                </div> }
            </div>,
        };

    componentDidMount() {
        this.rowDOMNode = ReactDOM.findDOMNode(this.rowNode.current);
        if (this.props.onFocus) {
            this.rowDOMNode?.addEventListener('mouseenter', this.handleMouseEnter);
        }
    }

    componentWillUnmount() {
        this.rowDOMNode?.removeEventListener('mouseenter', this.handleMouseEnter);
    }

    handleMouseEnter = (e: any) => {
        this.props.onFocus(this.props.index);
    }

    shouldComponentUpdate(nextProps: DataRowProps<TItem, TId> & FlexRowProps) {
        const eq = isEqual(this.props, nextProps);
        return !eq;
    }

    render() {
        const clickHandler = this.props.onSelect || this.props.onFold || this.props.onCheck;
        return <FlexRow
            onClick={ clickHandler && (() => clickHandler(this.props)) }
            rawProps={ {
                role: 'option',
                'aria-setsize': this.props.columns?.length == undefined ? undefined : this.props.columns.length + 1,
                'aria-posinset': this.props.index,
                'aria-checked': this.props.isChecked,
                ...(this.props.isSelectable && { 'aria-selected': this.props.isSelected }),
                ...this.props.rawProps,
            } }
            ref={ this.rowNode }
            cx={ [
                css.pickerRow,
                clickHandler && uuiMarkers.clickable,
                clickHandler && this.props.isFocused && uuiMod.focus,
                this.props.cx,
            ] }
        >
            <DataTableCell
                key='name'
                size={ this.props.size || '36' }
                padding={ this.props.padding || '12' }
                isFirstColumn={ true }
                isLastColumn={ false }
                column={ this.column }
                rowProps={ this.props }
                alignActions={ this.props.alignActions || 'top' }
            />
        </FlexRow>;
    }
}
