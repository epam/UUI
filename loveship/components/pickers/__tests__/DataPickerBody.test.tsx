import React from 'react';
import renderer from 'react-test-renderer';
import { renderWithContextAsync } from '@epam/test-utils';
import { dataSource } from './dataMocks';
import { DataPickerBody } from '../DataPickerBody';
import { DataPickerRow } from '../DataPickerRow';

describe('DataPickerBody', () => {
    const value = {};
    const onValueChange = jest.fn();
    const search = {
        value: 'test',
        onValueChange: jest.fn(),
    };

    it('should be rendered correctly', () => {
        const view = dataSource.getView({ topIndex: 0, visibleCount: 10, focusedIndex: 0 }, () => {});

        const tree = renderer
            .create(<DataPickerBody value={value} onValueChange={onValueChange} rows={view.getVisibleRows()} search={search} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    const rows = dataSource.props.items.map(props => (
        <DataPickerRow key={props.id} renderItem={item => <div>{item}</div>} id={props.id} rowKey={props.gender} index={props.id} />
    ));

    it('should be rendered correctly with extra props', async () => {
        const tree = await renderWithContextAsync(
            <DataPickerBody
                value={value}
                onValueChange={onValueChange}
                rows={rows}
                search={search}
                showSearch
                maxHeight={200}
                renderNotFound={() => <div>Not found</div>}
                editMode="dropdown"
                onKeyDown={jest.fn()}
                scheduleUpdate={jest.fn()}
                searchSize="24"
            />
        );
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with empty rows', () => {
        const tree = renderer.create(<DataPickerBody value={value} onValueChange={onValueChange} rows={[]} search={search} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
