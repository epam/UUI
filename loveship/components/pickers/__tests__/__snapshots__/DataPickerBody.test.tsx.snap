// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataPickerBody should be rendered correctly 1`] = `
<div
  className="root size-36 background-white spacing-6 container align-items-center"
  style={
    Object {
      "maxHeight": undefined,
    }
  }
>
  <div
    className=""
    onClick={[Function]}
    style={
      Object {
        "alignSelf": undefined,
        "flexBasis": "0",
        "flexGrow": 1,
        "flexShrink": undefined,
        "minWidth": 0,
        "textAlign": "center",
      }
    }
  >
    <div
      className="text font-sans color-night700 line-height-18 font-size-14 v-padding-9 container"
    >
      No Results found
    </div>
  </div>
</div>
`;

exports[`DataPickerBody should be rendered correctly with empty rows 1`] = `
<div
  className="root size-36 background-white spacing-6 container align-items-center"
  style={
    Object {
      "maxHeight": undefined,
    }
  }
>
  <div
    className=""
    onClick={[Function]}
    style={
      Object {
        "alignSelf": undefined,
        "flexBasis": "0",
        "flexGrow": 1,
        "flexShrink": undefined,
        "minWidth": 0,
        "textAlign": "center",
      }
    }
  >
    <div
      className="text font-sans color-night700 line-height-18 font-size-14 v-padding-9 container"
    >
      No Results found
    </div>
  </div>
</div>
`;

exports[`DataPickerBody should be rendered correctly with extra props 1`] = `
Array [
  <div
    className=""
  >
    <div
      className=""
      onClick={[Function]}
      style={
        Object {
          "alignSelf": undefined,
          "flexBasis": "0",
          "flexGrow": 1,
          "flexShrink": undefined,
          "minWidth": 0,
          "textAlign": undefined,
        }
      }
    >
      <div
        className="container uui-input-box -clickable colorSky root size-24 mode-form"
        onBlur={[Function]}
        onFocus={[Function]}
        tabIndex={-1}
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
            height={24}
            viewBox="0 0 12 24"
            width={12}
          >
            <use
              xlinkHref="#search-12.svg"
            />
          </svg>
        </div>
        <input
          aria-invalid={false}
          className="uui-input line-height-18 font-size-12 v-padding-2"
          inputMode="search"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Search"
          type="search"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="root size-36 background-white spacing-6 container align-items-center"
    style={
      Object {
        "maxHeight": 200,
      }
    }
  >
    <div>
      Not found
    </div>
  </div>,
]
`;
